src/lib/ReleaseHelper.ts

```diff
@@ -0,0 +1,182 @@
+import { Constants, Providers } from 'shared/constants';
+import { PropertiesOf } from 'shared/types';
+
+import { IVersion, VersionImpl } from './Version';
+
+export interface ReleaseHelperConfig {
+  currentReleaseVersion: string;
+  availableReleases: IReleases;
+  provider: PropertiesOf<typeof Providers>;
+  ignorePreReleases?: boolean;
+  isAdmin?: boolean;
+}
+
+export class ReleaseHelper {
```

Comment:
This class now contains all the release-specific logic (e.g. which versions are not upgradable, which version are skipped etc.). In a separate PR, I will migrate all the existing version checks to use this class.